### PR TITLE
update tap to latest to run test locally

### PR DESCRIPTION
**Pull request by sify21**
Signed-off-by: 司芳源 <sify1221@gmail.com>

I can not run tests locally using node version `v16.8.0`. 
- changed tap's version specifier to allow using latest version.
- changed `t.done()` to `t.end()` in `test/functions/coerce.js`

There are still 3 failed tests, I'm not familiar with Promise and don't know how to fix them.
```
 FAIL  test/bin/semver.js 3 failed of 4 821.243ms
 ✖ child test left in queue: t.test help output
 ✖ child test left in queue: t.test sorting and filtering
 ✖ child test left in queue: t.test coercing
```

## References
Related to https://github.com/tapjs/node-tap/issues/746

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "postpublish": "git push origin --follow-tags"
   },
   "devDependencies": {
-    "tap": "^14.10.7"
+    "tap": ">14.10.7"
```
Comment (ljharb) [inline package.json:14]: You never want to use > - the ^ already means “use the latest of v14.x”

Comment (sify21) [reply package.json:14]: the issue is not fixed until 15.0.4. maybe I should use ^15.0.4

**Comment by FEARTROLL** [comment]
Mw[]()

**Comment by lukekarrys** [comment]
Closing in favor of #433

**Pull request closed without merging**